src/domain/usecases/GetMasterLogFrameUseCase.ts

```diff
@@ -0,0 +1,11 @@
+import { MasterLogFrame } from "$/domain/entities/MasterLogFrame";
+import { FutureData } from "$/domain/entities/generic/Future";
+import { MasterLogFrameRepository } from "$/domain/repositories/MasterLogFrameRepository";
+
+export class GetMasterLogFrameUseCase {
+    constructor(private masterLogFrameRepository: MasterLogFrameRepository) {}
+
+    execute(codes: string[]): FutureData<MasterLogFrame[]> {
```

Comment:
When use case is called, it won't be very clear what are the strings being passed. I'd use an `options` here.